src/koheesio/spark/__init__.py

```diff
@@ -44,7 +51,7 @@ class Output(StepOutput):
     @property
     def spark(self) -> Optional[SparkSession]:
         """Get active SparkSession instance"""
-        return SparkSession.getActiveSession()
+        return self.spark_ or SparkSession.getActiveSession()
```

Comment:
Why not to do with getter and setter functionality ? 
```python
self.__spark=SparkSession.getActiveSession()
@property
def spark(self) -> Optional[SparkSession]:
    """Get active SparkSession instance"""
      return self.__spark

def spark
@spark.setter
def spark(self,spark_session:SparkSession):
    self._spark== spark_session
```

Reply:
because pydantic can take care of getting and setting. Less code and keeps things in scope of Pydantic. If a user passes an invalid SparkSession, it would automatically get errored out essentially

Reply:
I made the current implementation 100% pydantic based - have a look